sips/sip-021/sip-021-nakamoto.md

```diff
@@ -0,0 +1,1085 @@
+
+
+<em>Figure 3: Tenures N+1 and N+2 are empty.  Stackers recover from this by
+requiring the next non-empty tenure to include the empty tenures'
+<code>TenureChange</code> transactions, to prove that these tenures were in fact
+empty and no blocks were signed for them.  Because the tenures are empty, the
+inventory bitmap would mark them as 0s -- there is no block data to fetch.</em>
+
+To recover from this, Stackers would still produce `TenureChange` transactions
+for the empty tenures, and require the start block of the first non-empty tenure
+to include them.  In other words, Stackers _always_ produce a `TenureChange`
+transaction for each tenure; it just might not be included by the tenure's
+sortition-selected miner.
+
+### Stacker Turnover
+
+A miner tenure change happens every Bitcoin block.  In addition, there are
+Stacker tenure changes, which happen once every 2100 Bitcoin blocks (one reward
+cycle).  In a Stacker tenure change, a new set of Stackers are selected by the
+PoX anchor block (see SIP-007).
+
+Because Stacks no longer forks, the PoX anchor block is always known 100 Bitcoin
+blocks prior to the start of the next reward cycle:  it is simply the the start
+block of the 2000th tenure in the current reward cycle.
+
+The PoX anchor block identifies the next Stackers.  They have 100 Bitcoin blocks
+to prepare for signing Stacks blocks.  Within this amount of time, the new
+Stackers would complete a WSTS DKG for signing blocks (among other things, such
+as the sBTC peg wallet hand-off).  The PoX contract will require Stackers to
+register their block-signing keys when they stack or delegate-stack STX, so the
+entire network knows enough information to validate their WSTS Schnorr
+signatures on blocks.
+
+### PoX Failure
+
+In the event that PoX does not activate, the chain halts.  If there are no
+Stackers, then block production cannot happen.
+
+### Changes to PoX
+
+To support tenure changes, this proposal calls for a new PoX contract, `.pox-4`.
+The `.pox-4` contract would be altered over the current PoX contract (`.pox-3`)
+to meet the following requirements:
+
+* Stackers register a WSTS signing key when they call `stack-stx` and optionally
+  `delegate-stx`.
+    * In `delegate-stx`, Stackers may instead elect for the delegate to supply
+      the signing key.  In this case, the delegate carries out the Stacker's
+responsibility for signing blocks.
+* The stacking minimum is abolished.  If there is at least one stacker that is
+  able to claim one reward slot, then PoX activates.
+* The `.pox-4` contract will expose each reward cycle's full reward set,
+  including the signing keys for each stacker, via a public function.
+Internally, the Stacks node will call a _private_ function to load the reward
+set into the `.pox-4` data space after it identifies the PoX anchor block.  This
+is required for other dependent SIPs, such as the sBTC SIP.
+
```

Comment:
```suggestion
* Stacks token holders can't vote to deactivate PoX. Function `reject-pox` is removed in `.pox-4`.
```

Reply:
I think that this probably should have been included so that it was explicit, but I think it's too late to add something like this now. We only want to add/change inconsequential things in the doc after approval.